Docs: Use history change basically everywhere (#11571)
PostHog/posthog.com: contents/tutorials/nextjs-app-directory-analytics.md

Reviewer: No mention of the new `instrumentation` file?

Author: Not yet, we need to test that and figure out how best to use it

First, we need a PostHog instance ([signup for free](https://app.posthog.com/signup)). From this instance, we need a project API key, which is in project settings, and an the relevant ingestion address (`http://us.i.posthog.com`, `http://eu.i.posthog.com`, or a custom address). Add both of these to a `.env.local` file in our base directory.

```
NEXT_PUBLIC_POSTHOG_KEY=<ph_project_api_key>
NEXT_PUBLIC_POSTHOG_HOST=<ph_client_api_host>
```

Using the Next.js app router requires us to initialize PostHog differently than with the [pages directory](/tutorials/nextjs-analytics). Specifically, the app router server-side renders components by default, and the `posthog-js` library is a client-side library.

To make these work together, create a `providers.js` file and set up the `PostHogProvider` with the `'use client'`  directive. Make sure to set `capture_pageview` to `history_change` because Next.js acts as a single-page app and doesn't fire the page load events PostHog usually relies to capture pageviews.

```js
// app/providers.js
'use client'
import posthog from 'posthog-js'
import { PostHogProvider } from 'posthog-js/react'
import { useEffect } from 'react'

> export function PHProvider({ children }) {

Author: The problem is that we are importing `import { PostHogProvider } from 'posthog-js/react'` which I find more confusing if they are named the same thing.

  useEffect(() => {
    posthog.init(process.env.NEXT_PUBLIC_POSTHOG_KEY, {
      api_host: process.env.NEXT_PUBLIC_POSTHOG_HOST,
      capture_pageview: 'history_change',
    })
  }, []);

  return <PostHogProvider client={posthog}>{children}</PostHogProvider>
}
```

Once we do this, we can then import the `provider.js` file in our `app/layout.js` file, and wrap our app in the PostHog provider. 

```js
// app/layout.js
import './globals.css'
import { PHProvider } from './providers'

export default function RootLayout({ children }) {
  return (
    <html lang="en">
      <PHProvider>
        <body>{children}</body>
      </PHProvider>
    </html>
  )
}
```

> After setting this up, events and pageviews start being autocaptured into your PostHog instance. This also means you can use the PostHog on the client side in all your client-side rendered Next.js components (the ones with the `"use client"` directive).

Reviewer 2: ```suggestion
After you set this up, PostHog starts autocapturing events and pageviews. This also means you can use the PostHog on the client side in all your client-side rendered Next.js components (the ones with the `"use client"` directive).
```

Nit:
